### ISSUE TITLE
Enable JPM-internal content on Salt site

### ISSUE THREAD

**Issue by james-nash** [jpmorganchase/salt-ds#1554]
### Business need
While the vast majority of content on the Salt website should always be public, we are catering to a large audience of consumers within JPM. There is additional content we would like to provide to that audience:

* It's likely that there will be company-internal components / patterns / examples / etc. that we'd like to list in our community indexes (see #1205), alongside any publicly-accessible ones.
* There may be company-specific guidance we'd like to provide to practitioners (e.g. how to setup a standard JPM dev environment to use Salt, how to request a Figma account, etc.) which are not applicable or relevant to people outside of JPM.
* There may be case-studies that relate to JPM-internal applications which use Salt which we'd like to share, but cannot do in public since the app they're about is not public.

### Possible approaches
There's a few ways we could potentially meet that need:

1. Create a separate JPM-internal sister website that contains all the extra content that can't go on the public Salt website
2. Create a JPM-internal version of the Salt site that is a clone of the public one but also includes additional internal content
3. Detect if a visitor to the Salt website is on the JPM network/VPN and display additional JPM-internal content alongside the usual public content

These are roughly in order of technical complexity. A completely separate site is likely to be easier to being able to generate 2 versions of the Salt site from the same source code, which in turn is probably easier than figuring out how to serve alternate versions of the site at the same URL.

However, in terms of user experience, I believe option 3 is the most convenient:

* Everyone, be they JPM staff or not, uses the exact same URL (`www.saltdesignsystem.com`), but JPM staff will see some extra content that others do not.
* This would also simplify comms and sharing since wherever we talk about Salt, we can always share the same URL - there'd be no need to continuously caveat it with "If you work at JPM go here ..., otherwise go here...".
* It also reduces the risk of people getting lost or confused because they accidentally visited the "wrong" link and cannot find content they were expecting to see.
* Allows for a more integrated reading experience, since JPM-internal content can be surfaced directly alongside public content as opposed to requiring the visitor to go to a completely different site to find that additional info.

**I therefore want us to explore what it would take to achieve option 3.** If a spike reveals that it's not possible or excessively complex, then we can explore options 1 & 2.

### Technical requirements
Assuming we can go with the preferred option (same URL, but serve up different content for JPM-internal and public visitors), we'd have the following technical requirements:

* Ability to show extra sections or alternate content within a page to JPM-internal visitors
    * In other words, we'd like to be able to do something like `if (jpm internal visitor ) { show secret content } else { show public content }`
* Ability to show additional pages to JPM-internal visitors
    * Which may also require additional links in the main site navigation to appear
* Ability to source internal-content from somewhere other than our public GitHub repo
    * Since our site's code is open-source, all of its contents are as well. Clearly, JPM-internal content can't reside in the same repo, so where does it go?
